src/builder.rs

```diff
@@ -391,6 +396,21 @@ fn prepare_header(dst: &mut Write, header: &mut Header, path: &Path) -> io::Resu
     Ok(())
 }
 
+fn prepare_header_link(dst: &mut Write, header: &mut Header, link_name: &Path) -> io::Result<()> {
+    // Same as previous function but for linkname
+    if let Err(e) = header.set_link_name(&link_name) {
+        let data = path2bytes(&link_name)?;
+        if data.len() < header.as_old().linkname.len() {
+            return Err(e);
+        }
+        let mut header2 = prepare_header(data.len() as u64, b'K');
+        let mut data2 = data.chain(io::repeat(0).take(0));
```

Comment:
I think this was copied from before (sorry for lots of comments like this) but I actually have no idea why `io::repeat(0).take(0)` is above (and copied down here). It seems like the `.chain` should be safe to delete, right?

Reply:
Again, not sure why this was here, `.take(1)` would have made sense for the null-terminated name I mentioned in the other comment. This isn't a problem when name is shorter than 512 since `append()` will pad with 0 but I don't know if this last null byte is mandatory. 
Thing is `append()` wants something that implements `Read` so we can't just delete the `.chain` and call the function with `data` directly, would a `Cursor` be better ?

Reply:
Hm let's actually switch this to `io::repeat(0).take(1)`, as I agree that was probably the original intent and it's been "accidentally working" so far otherwise.

Reply:
alright